### PR TITLE
Move remaining t-channel code from core to tchannel package

### PR DESCRIPTION
**Pull request by pavolloffay**
Related to #2129 

This PR removes `tchannel-go` references from the code components. The only remaining references are in the hotrod app which will be removed in a follow-up PR.

```
git grep tchannel-go/
cmd/opentelemetry/go.sum:1233:github.com/uber/tchannel-go v1.10.0 h1:YOihLHuvkwT3nzvpgqFtexFW+pb5vD1Tz7h/bIWApgE=
cmd/opentelemetry/go.sum:1234:github.com/uber/tchannel-go v1.10.0/go.mod h1:Rrgz1eL8kMjW/nEzZos0t+Heq0O4LhnUJVA32OvWKHo=
cmd/opentelemetry/go.sum:1235:github.com/uber/tchannel-go v1.16.0 h1:B7dirDs15/vJJYDeoHpv3xaEUjuRZ38Rvt1qq9g7pSo=
cmd/opentelemetry/go.sum:1236:github.com/uber/tchannel-go v1.16.0/go.mod h1:Rrgz1eL8kMjW/nEzZos0t+Heq0O4LhnUJVA32OvWKHo=
examples/hotrod/services/driver/client.go:23:   "github.com/uber/tchannel-go"
examples/hotrod/services/driver/client.go:24:   "github.com/uber/tchannel-go/thrift"
examples/hotrod/services/driver/server.go:21:   "github.com/uber/tchannel-go"
examples/hotrod/services/driver/server.go:22:   "github.com/uber/tchannel-go/thrift"
examples/hotrod/services/driver/thrift-gen/driver/tchan-driver.go:10:   "github.com/uber/tchannel-go/thrift"
go.mod:75:      github.com/uber/tchannel-go v1.16.0
go.sum:436:github.com/uber/tchannel-go v1.16.0 h1:B7dirDs15/vJJYDeoHpv3xaEUjuRZ38Rvt1qq9g7pSo=
go.sum:437:github.com/uber/tchannel-go v1.16.0/go.mod h1:Rrgz1eL8kMjW/nEzZos0t+Heq0O4LhnUJVA32OvWKHo=
tchannel/agent/app/configmanager/tchannel/manager.go:21:        "github.com/uber/tchannel-go"
tchannel/agent/app/configmanager/tchannel/manager.go:22:        "github.com/uber/tchannel-go/thrift"
tchannel/agent/app/configmanager/tchannel/manager_test.go:24:   "github.com/uber/tchannel-go/thrift"
tchannel/agent/app/reporter/tchannel/builder.go:23:     tchannel "github.com/uber/tchannel-go"
tchannel/agent/app/reporter/tchannel/builder_test.go:24:        "github.com/uber/tchannel-go"
tchannel/agent/app/reporter/tchannel/reporter.go:21:    "github.com/uber/tchannel-go"
tchannel/agent/app/reporter/tchannel/reporter.go:22:    "github.com/uber/tchannel-go/thrift"
tchannel/agent/app/testutils/mock_collector.go:22:      "github.com/uber/tchannel-go"
tchannel/agent/app/testutils/mock_collector.go:23:      "github.com/uber/tchannel-go/thrift"
tchannel/agent/app/testutils/mock_collector_test.go:24: "github.com/uber/tchannel-go/thrift"
tchannel/collector/app/collector.go:18: "github.com/uber/tchannel-go"
tchannel/collector/app/handler/tchannel_handler.go:18:  "github.com/uber/tchannel-go/thrift"
tchannel/collector/app/sampling/handler.go:18:  "github.com/uber/tchannel-go/thrift"
tchannel/collector/app/server/thrift.go:21:     "github.com/uber/tchannel-go"
tchannel/collector/app/server/thrift.go:22:     "github.com/uber/tchannel-go/thrift"
tchannel/pkg/discovery/peerlistmgr/peer_list_mgr.go:25: "github.com/uber/tchannel-go"
tchannel/pkg/discovery/peerlistmgr/peer_list_mgr_test.go:27:    tchannel "github.com/uber/tchannel-go"
tchannel/pkg/discovery/peerlistmgr/peer_list_mgr_test.go:28:    "github.com/uber/tchannel-go/raw"
tchannel/pkg/discovery/peerlistmgr/peer_list_mgr_test.go:29:    "github.com/uber/tchannel-go/testutils"
thrift-gen/baggage/tchan-baggage.go:10: "github.com/uber/tchannel-go/thrift"
thrift-gen/jaeger/tchan-jaeger.go:10:   "github.com/uber/tchannel-go/thrift"
thrift-gen/sampling/tchan-sampling.go:10:       "github.com/uber/tchannel-go/thrift"
thrift-gen/zipkincore/tchan-zipkincore.go:10:   "github.com/uber/tchannel-go/thrift"
```

```
git grep tchannel/
tchannel/agent/app/configmanager/tchannel/manager_test.go:26:   "github.com/jaegertracing/jaeger/tchannel/agent/app/testutils"
tchannel/agent/app/reporter/tchannel/builder.go:27:     "github.com/jaegertracing/jaeger/tchannel/pkg/discovery/peerlistmgr"
tchannel/agent/app/reporter/tchannel/collector_proxy.go:23:     "github.com/jaegertracing/jaeger/tchannel/agent/app/configmanager/tchannel"
tchannel/agent/app/reporter/tchannel/collector_proxy_test.go:28:        "github.com/jaegertracing/jaeger/tchannel/agent/app/configmanager/tchannel"
tchannel/agent/app/reporter/tchannel/reporter.go:25:    "github.com/jaegertracing/jaeger/tchannel/pkg/discovery/peerlistmgr"
tchannel/agent/app/reporter/tchannel/reporter_test.go:27:       "github.com/jaegertracing/jaeger/tchannel/agent/app/testutils"
tchannel/collector/app/collector.go:23: "github.com/jaegertracing/jaeger/tchannel/collector/app/server"
tchannel/collector/app/server/thrift.go:27:     tHandler "github.com/jaegertracing/jaeger/tchannel/collector/app/handler"
tchannel/collector/app/server/thrift.go:28:     "github.com/jaegertracing/jaeger/tchannel/collector/app/sampling"

```

### CODE REVIEW

File: cmd/agent/app/builder_test.go [hunk 1]
```diff
@@ -221,10 +215,8 @@ func TestCreateCollectorProxy(t *testing.T) {
 
 	for _, test := range tests {
 		flags := &flag.FlagSet{}
-		tchannel.AddFlags(flags)
 		grpc.AddFlags(flags)
 		reporter.AddFlags(flags)
-		app.AddFlags(flags)
```
Comment (objectiser) [inline cmd/agent/app/builder_test.go:227]: Why are these flags being removed?

File: cmd/agent/app/processors/thrift_processor_test.go [hunk 2]
```diff
@@ -181,12 +189,12 @@ func assertJaegerProcessorCorrectness(t *testing.T, collector *testutils.MockTCo
 	assertProcessorCorrectness(t, metricsFactory, sizeF, nameF, "jaeger")
 }
 
-func assertZipkinProcessorCorrectness(t *testing.T, collector *testutils.MockTCollector, metricsFactory *metricstest.Factory) {
+func assertZipkinProcessorCorrectness(t *testing.T, collector *testutils.GrpcCollector, metricsFactory *metricstest.Factory) {
 	sizeF := func() int {
-		return len(collector.GetZipkinSpans())
+		return len(collector.GetJaegerBatches())
```
Comment (objectiser) [inline cmd/agent/app/processors/thrift_processor_test.go:194]: Why have these changed to Jaeger batches when in a zipkin processor test? Is zipkin format collection via the agent relevant now? If not, then we should probably change the name of the test.

File: cmd/agent/app/builder_test.go (hunk 1 above)
Comment (pavolloffay) [reply cmd/agent/app/builder_test.go:227]: leftovers, not needed anymore

File: cmd/agent/app/processors/thrift_processor_test.go (hunk 2 above)
Comment (pavolloffay) [reply cmd/agent/app/processors/thrift_processor_test.go:194]: Jaeger agent can injest zipkin thrift spans https://www.jaegertracing.io/docs/1.18/deployment/#agent. However the reporter in the agent convert these spans to jaeger batches.

Comment (objectiser) [reply cmd/agent/app/processors/thrift_processor_test.go:194]: ok

File: cmd/collector/app/span_processor.go [hunk 3]
```diff
@@ -164,7 +164,7 @@ func (sp *spanProcessor) ProcessSpans(mSpans []*model.Span, options processor.Sp
 	for i, mSpan := range mSpans {
 		ok := sp.enqueueSpan(mSpan, options.SpanFormat, options.InboundTransport)
 		if !ok && sp.reportBusy {
-			return nil, tchannel.ErrServerBusy
+			return nil, fmt.Errorf("server busy")
```
Comment (yurishkuro) [inline cmd/collector/app/span_processor.go:167]: nit: this should be a dedicated error (similar to TraceNotFound in the query service) and the gRPC handler should translate it it to gRPC-specific status code. But it's not critical as there's no special processing in the agent for this.

Comment (yurishkuro) [reply cmd/collector/app/span_processor.go:167]: Actually, might be still critical for us because "busy" is understood by the service mesh

Comment (pavolloffay) [reply cmd/collector/app/span_processor.go:167]: PR https://github.com/jaegertracing/jaeger/pull/2314

**Pull request merged**